lisp/ess-r-mode.el

```diff
@@ -2203,6 +2202,71 @@ state.")
 
 
 
+;;;*;;; Inferior R mode
+
+(defvar inferior-ess-r-mode-map
+  (let ((map (make-sparse-keymap)))
+    (define-key map "\M-\r" 'ess-dirs)
+    map)
+  "Keymap for `inferior-ess-r-mode'.")
+
+;; TOTHINK: Prevent string delimiting characters from messing up output in the
+;; inferior buffer
+(defvar inferior-ess-r-mode-syntax-table
```

Comment:
I forgot how syntax tables work with derived modes. Is there a notion of derived syntax table? Don't we have to change our strategy of installing syntax tables now?

Reply:
> derived syntax table

Not really. You can still do `copy-syntax-table` to sort-of derive from a specific syntax table. 

> Don't we have to change our strategy of installing syntax tables now?

Yes. We no longer rely on the various -customize-alists (I thought I had removed all these but I see now that I missed a few. Will fix that now). Instead the syntax tables are activated by the major modes.

Reply:
> We no longer rely on the various -customize-alists (I thought I had removed all these but I see now that I missed a few. 

Well, this is a very big step which will ruin quite a lot of user configs out there. But most importantly this might not be the right approach. As I said earlier, one important function of that customize-alist is to clone modes to external buffers like help,  preview etc. which are generic across dialects and for which derivation is not an option. I don't think we ever discuss this change.

Reply:
Do you think many people change the syntax table of alist variables via their user config? Or is it something else?

Reply:
Not syntax table, but other variables for sure. I don't know why that variable was called `-customize-` alist and if it was ever intended to be `customized` or was documented as being `customizable`. To me it was always part of internals and I don't think non-lisp-aware people ever touched it. So I think if we break and remove it in the next release those guys who do will find a quick way to fix their configs.